packages/open-next/src/utils/promise.ts

```diff
@@ -58,3 +58,64 @@ export class DetachedPromiseRunner {
     });
   }
 }
+
+async function awaitAllDetachedPromise() {
+  const promisesToAwait =
+    globalThis.__openNextAls.getStore()?.pendingPromiseRunner.await() ??
+    Promise.resolve();
+  if (globalThis.openNextWaitUntil) {
+    globalThis.openNextWaitUntil(promisesToAwait);
+    return;
+  }
+  await promisesToAwait;
+}
+
+function provideNextAfterProvider() {
+  /** This should be considered unstable until `unstable_after` is stablized. */
+  const NEXT_REQUEST_CONTEXT_SYMBOL = Symbol.for("@next/request-context");
+
+  // This is needed by some lib that relies on the vercel request context to properly await stuff.
+  // Remove this when vercel builder is updated to provide '@next/request-context'.
+  const VERCEL_REQUEST_CONTEXT_SYMBOL = Symbol.for("@vercel/request-context");
+
+  const openNextStoreContext = globalThis.__openNextAls.getStore();
+
+  const awaiter =
```

Comment:
nit: rename this to `waitUntil` and change below `get: () => ({ waitUntil })` - I feel like introducing a new name is not worth